webapp/tests/karma/ts/modules/messages/messages.component.spec.ts

```diff
@@ -181,4 +202,11 @@ describe('Messages Component', () => {
 
     expect(spySubscriptionsUnsubscribe.callCount).to.equal(1);
   });
+
+  it('it should retrieve the hierarchy level of the connected user', () => {
```

Comment:
Should include tests to cover the various new logic.  Make sure that the current lineage level for the user is filtered out from the displayed lineage (but only when the user is offline).

Reply:
+1
every object in `conversations` to have set the right lineage value in different possible scenarios